### PR TITLE
🐛 Support `deployments/scale` in root compute

### LINKED ISSUE #2261: bug: root:compute maximal permission policy doesn’t grant access to the deployments/scale resource
### Describe the bug

The maximal permission policy of the `root:compute` workspace doesn’t grant access to the `deployments/scale` resource.

### Steps To Reproduce

```bash
$ k scale deployment kuard --replicas=0
Error from server (Forbidden): deployments.apps "kuard" is forbidden: User "kcp-admin" cannot patch resource "deployments/scale" in API group "apps" in the namespace "default"
```

### Expected Behaviour

Be able to scale the deployment

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by davidfestal**
## Summary

Add support for the `deployments/scale` sub-resource in root compute.
The maximal permission policy in the `root:compute` workspace didn’t grant access to the `deployments/scale` resource

## Related issue(s)

Fixes #2261

### CODE REVIEW

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go [hunk 1]
```diff
@@ -153,4 +163,80 @@ func TestRootComputeWorkspace(t *testing.T) {
 		}
 		return true, ""
 	}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+	t.Logf("Wait for being able to list Deployments in the user workspace")
+	require.Eventually(t, func() bool {
```
Comment (ncdc) [inline test/e2e/reconciler/locationworkspace/rootcompute_test.go:168]: We usually try to use `framework.Eventually`. Was there a specific reason not to use it here?

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go [hunk 2]
```diff
@@ -153,4 +163,80 @@ func TestRootComputeWorkspace(t *testing.T) {
...
+		_, err := kubeClusterClient.Cluster(consumerWorkspace).AppsV1().Deployments("default").List(ctx, metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			t.Logf("service err %v", err)
+			return false
+		} else if err != nil {
+			t.Logf("service err %v", err)
+			return false
+		}
```
Comment (ncdc) [inline test/e2e/reconciler/locationworkspace/rootcompute_test.go:176]: Given there is no different behavior for different types of errors, I would recommend collapsing this down to only checking for an error

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go (hunk 1 above)
Comment (davidfestal) [reply test/e2e/reconciler/locationworkspace/rootcompute_test.go:168]: I just copied the code above TBF :-)

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go (hunk 2 above)
Comment (davidfestal) [reply test/e2e/reconciler/locationworkspace/rootcompute_test.go:176]: Copy of the above code as well ;-)

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go (hunk 1 above)
Comment (davidfestal) [reply test/e2e/reconciler/locationworkspace/rootcompute_test.go]: fixed

File: test/e2e/reconciler/locationworkspace/rootcompute_test.go (hunk 2 above)
Comment (davidfestal) [reply test/e2e/reconciler/locationworkspace/rootcompute_test.go]: fixed

**Review by ncdc — APPROVED** [review]

**Pull request merged**